hack/ci-test-e2e.sh

```
@@ -8,20 +8,71 @@ set -o monitor
+  # Ginkgo tests generate junit-ginkgo.xml directly via reporters.NewJUnitReporter
+  # (configured in test/e2e/e2e_suite_test.go)
+  # This file is created automatically when Ginkgo specs run
+
+  # Merge all junit-*.xml files into final junit.xml
+  # This handles both native and Ginkgo tests, working correctly at any migration stage
+  if ! command -v bin/junitmerge &> /dev/null; then
+    echo "Error: bin/junitmerge not found. This should be built by 'make e2e'." >&2
+    exit 1
+  fi
+
+  if ! bin/junitmerge $(find "${ARTIFACT_DIR}" -name "junit-*.xml" 2>/dev/null || true) > "${ARTIFACT_DIR}/junit.xml"; then
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)